### DISCUSSION TITLE
why using await for setting/getting sessions ?

### DISCUSSION THREAD

**Discussion by HosMercury** [maxcountryman/tower-sessions#151] [Q&A] [answered]
although sometimes it could be simple number or non-async stuff.

**Comment by maxcountryman** [comment, marked as answer]
This is necessary because sessions are lazy and therefore are not loaded until the moment they're actually used; loading a session from some arbitrary store is an asynchronous action. The motivation for this design was outlined in depth in [this PR](https://github.com/maxcountryman/tower-sessions/pull/112) and the linked discussion therein.
